pkg/kubelet/dockertools/manager.go

```diff
@@ -1401,6 +1401,12 @@ func (dm *DockerManager) computePodContainerChanges(pod *api.Pod, runningPod kub
 			containersToKeep[containerID] = index
 			continue
 		}
+
+		// Record an event for the healthcheck failure.
+		ref, err := kubecontainer.GenerateContainerRef(pod, &container)
+		if err == nil {
+			dm.recorder.Eventf(ref, "failed", "Healthcheck failed with probe result: %v", result)
+		}
```

Comment:
We also record events sometimes in `prober.Probe`. We should do it in one place.

It'd also be good to distinguish between liveness check and readiness check for debugging (as mentioned in #8238). Perhaps we should consolidate the two PR? What do you think?